nixos/modules/services/databases/pgbouncer.nix

```diff
@@ -128,7 +135,181 @@ in
     };
 
     settings = lib.mkOption {
-      type = settingsFormat.type;
+      type = lib.types.submodule {
```

Comment:
I didn't have the concentration to proofread all the options but generally looked okay to me.